dataland-csvconverter/src/main/kotlin/org/dataland/csvconverter/CsvToJsonConverter.kt

```diff
@@ -0,0 +1,222 @@
+     * Method to get a list of CompanyInformation objects generated from the csv file
+     */
+    fun buildListOfCompanyInformation(): List<CompanyInformation> {
+        return rawCsvData.filter { validateLine(it) }.map {
+            CompanyInformation(
+                companyName = getValue(columnMapping["companyName"]!!, it),
+                headquarters = getValue(columnMapping["headquarters"]!!, it),
+                sector = getValue(columnMapping["sector"]!!, it),
+                marketCap = getMarketCap(columnMapping["marketCap"]!!, it),
+                reportingDateOfMarketCap = LocalDate.parse(
+                    getValue(columnMapping["reportingDateOfMarketCap"]!!, it),
+                    DateTimeFormatter.ofPattern("d.M.yyyy")
+                ),
+                identifiers = getIdentifiers(it),
+                indices = getStockIndices(it)
+            )
+        }
+    }
+
+    private fun validateLine(csvLineData: Map<String, String>): Boolean {
+        // Skip all lines with financial companies or without market cap
+        return !(
+            getValue(columnMapping["companyType"]!!, csvLineData) in listOf("FS", notAvailableString) ||
+                getValue(columnMapping["marketCap"]!!, csvLineData) == notAvailableString
+            )
+    }
+
+    private fun getValue(columnName: String, csvData: Map<String, String>): String {
+        return csvData[columnName]!!.trim().ifBlank {
+            notAvailableString
+        }
+    }
+
+    private fun getMarketCap(columnHeader: String, csvLineData: Map<String, String>): BigDecimal {
+        return getScaledValue(columnHeader, csvLineData, euroUnitConverter)!!
+    }
+
+    private fun getScaledValue(columnHeader: String, csvData: Map<String, String>, scaleFactor: String): BigDecimal? {
+        // The numeric value conversion assumes "," as decimal separator and "." to separate thousands
+        return getValue(columnHeader, csvData).replace("[^,\\d]".toRegex(), "").replace(",", ".")
+            .toBigDecimalOrNull()?.multiply(scaleFactor.toBigDecimal())
+    }
+
+    private fun getIdentifiers(csvLineData: Map<String, String>): List<CompanyIdentifier> {
+        return identifierMapping.keys.map {
+            CompanyIdentifier(identifierValue = getValue(identifierMapping[it]!!, csvLineData), identifierType = it)
+        }.filter { it.identifierValue != notAvailableString }
+    }
+
+    private fun getStockIndices(csvLineData: Map<String, String>): Set<StockIndex> {
+        return stockIndexMapping.keys.filter { csvLineData[stockIndexMapping[it]]!!.isNotBlank() }.toSet()
+    }
+
+    /**
+     * Method to get a list of CompanyAssociatedEuTaxonomyData objects generated from the csv file
+     */
+    fun buildListOfEuTaxonomyData(): List<CompanyAssociatedData<EuTaxonomyData>> {
+        return rawCsvData.filter { validateLine(it) }.withIndex().map { (index, csvLineData) ->
+            CompanyAssociatedData(
+                companyId = "${index + 1}",
```

Comment:
uh - das ist aber auch scharf. 
Also - wir machen eine Liste von Companies. Wir machen eine Liste von CompanyAssociatedData. Und weil wir wissen, dass wir die Liste in gleicher Reihenfolge hochladen werden, und wir wissen, dass wir die companyId bei 1 beginnen zu zählen, können wir hier schon die companyId vergeben.

Eigentlich stößt uns das doch wieder auf das Problem, dass wir das hochladen von CompanyAssociatedData nicht mit einem (technischen) schlüssel verknüpfen sollten, oder?

Reply:
In der Tat greifen die Tests gar nicht auf die companyId zu sondern setzen dort immer die tatsächlich vom System vergebene ein. Aber ich stimme dir zu das die Verknütpfung von Daten und Company über eine ID die erst bei Laufzeit verfügbar ist, unglücklich ist. Das sollten wir mal überdenken.

Reply:
Das müsste auch spätestens nach der EuroDaT-Integration zu einem echten Problem werden wenn wir die Daten zum Beispiel auf Preview hochladen.

Reply:
dann müssen wir es wohl da lösen :-)

Aber hier schließe ich dann mal den Issue.